packages/shared/lib/core/network/classes/stardust-network.class.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import { get } from 'svelte/store'
+
+import { activeProfile, updateActiveProfile } from '@core/profile/stores'
+
+import { ChainType } from '../enums'
+import { IChain, IIscpChainConfiguration, INetwork, INetworkStatus } from '../interfaces'
+import { networkStatus } from '../stores'
+import { ChainConfiguration, NetworkMetadata } from '../types'
+
+import { IscpChain } from './iscp-chain.class'
+
+export class StardustNetwork implements INetwork {
+    private readonly _metadata: NetworkMetadata
+    private readonly _chains: IChain[]
+
+    constructor(metadata: NetworkMetadata, chainConfigurations: ChainConfiguration[]) {
+        this._metadata = metadata
+        this._chains = this.constructChains(chainConfigurations)
+    }
+
+    private constructChains(chainConfigurations: ChainConfiguration[]): IChain[] {
+        return chainConfigurations.map((chainConfiguration) => {
+            switch (chainConfiguration.type) {
+                case ChainType.Iscp:
+                    return new IscpChain(chainConfiguration)
+                case ChainType.Evm:
+                    return undefined
+                default:
+                    return undefined
+            }
+        })
+    }
+
+    getMetadata(): NetworkMetadata {
+        return this._metadata
+    }
+
+    getStatus(): INetworkStatus {
+        return get(networkStatus) ?? {}
+    }
+
+    getChain(chainId: number): IChain {
+        return this._chains.find((chain) => chain.getConfiguration().chainId === chainId)
+    }
+
+    getChains(): IChain[] {
+        return this._chains
+    }
```

Comment:
we can transform these getters into "real" getters in javascript style (ref: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get?retiredLocale=de)

```suggestion
    get metadata(): NetworkMetadata {
        return this._metadata
    }

    get status(): INetworkStatus {
        return get(networkStatus) ?? {}
    }

    get chain(chainId: number): IChain {
        return this._chains.find((chain) => chain.getConfiguration().chainId === chainId)
    }

    get chains(): IChain[] {
        return this._chains
    }
```

Reply:
Yes that makes sense, although probably only for the non-async variables. Did some looking around it seems async getters are possible, but I think that'd mean we have to use it like `const metadata = await chain.metadata`, which is sorta strange for a property.

Reply:
I think it may look strange to have both usages for getting data alongside eachother (some require it because it takes a parameter). I'd prefer just to keep it consistent, but we can change it later if we need to make use of a getter method.